solutions/ccc-Twenty-four.mdx

```diff
@@ -0,0 +1,118 @@
+---
+id: ccc-Twenty-four
+source: CCC 2008
+title: Twenty-four
+author: Paul Chen
+---
+
+## Explanation
+
+We can complete search through all possible ways to create an arithmetic expression with a card hand. 
+For each permutation of the card hand, we try every way of placing arithmetic operators. 
+Note that there are two possible ways that parentheses can be arranged: `((( ) ) )`, or `(( ) ( ))`, so we must try both cases.
+The arithmetic expression with the maximum value that is not greater than 24 is our answer.
+
+## Implementation
+
+<LanguageSection>
+
+<CPPSection>
+```cpp
+#include <bits/stdc++.h>
+
+using namespace std;
+
+// Function that takes in two numbers and an operation and returns the result.
+int operation(int op, int num1, int num2) {
+	switch (op) {
+		case 0:
+			return num1 + num2;
+		case 1:
+			return num1 - num2;
+		case 2:
+			return num1 * num2;
+		case 3: {
+			// The divisor cannot be 0 and the quotient must be a whole number.
+			if (num2 == 0 || num1 % num2) {
+				return INT_MIN;
```

Comment:
no, don't use INT_MIN. Create a small constant (-1e9 or smth) and use that instead. INT_MIN is prone to overflow errors, and is bad practice.

Reply:
no??? if so give an SO link
if you're doing things right this shouldn't overflow

Reply:
oh ok well I'll leave it as-is for now